client/subscription-edit-page/index.tsx

```diff
@@ -0,0 +1,280 @@
+/* eslint-disable prettier/prettier */
+/* global jQuery */
+
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { createRoot } from 'react-dom/client';
+import { __ } from '@wordpress/i18n';
+
+// TypeScript declaration for jQuery
+declare const jQuery: (
+	selector: any
+) => {
+	on: ( event: string, handler: () => void ) => void;
+	off: ( event: string, handler: () => void ) => void;
+};
+
+/**
+ * Internal dependencies
+ */
+import type {
+	PaymentMethodSelectProps,
+	WCPayPMSelectorData,
+	FetchUserTokensResponse,
+} from './types';
+import UserTokenCache from './user-token-cache';
+
+/**
+ * Add a listener to the customer select.
+ *
+ * This could be a shorter method, but because the customer select
+ * element uses select2, it does not emit the typical `change` event.
+ *
+ * @param {(userId: number) => void} callback The callback to call when the customer is changed.
+ * @return {() => void} The cleanup function.
+ */
+const addCustomerSelectListener = (
+	callback: ( userId: number ) => void
+): ( () => void ) => {
+	const customerUserSelect = document.getElementById(
+		'customer_user'
+	) as HTMLSelectElement | null;
```

Comment:
(minor) rather than using `as`, we can check the type returned by the `getElementById`. This way we don't need to force the casting of the type.

```suggestion
	const element = document.getElementById( 'customer_user' );
	const customerUserSelect =
		element instanceof HTMLSelectElement ? element : null;
```

Reply:
Addressed in https://github.com/Automattic/woocommerce-payments/pull/11143/commits/4043b4914c6f4e20531c154d2396f124212d7fe1